instructionAPI/src/InstructionDecoder-power.C

```diff
@@ -294,28 +294,158 @@ which are both 0).
+
+          // Indirectly branch through the link register
+          auto lr = makeRegisterExpression(ppc32::lr);
+          constexpr bool is_call = false;
+          constexpr bool is_indirect = true;
+          constexpr bool is_conditional = false;
+          constexpr bool is_fallthrough = false;
+          insn_in_progress->addSuccessor(lr, is_call, is_indirect, is_conditional, is_fallthrough);
+        }
+      }
 
       if(bcIsConditional) {
-        insn_in_progress->addSuccessor(makeFallThroughExpr(), false, false, false, true);
+        auto target = makeFallThroughExpr();
+        constexpr bool is_call = false;
+        constexpr bool is_indirect = false;
+        constexpr bool is_conditional = false;
+        constexpr bool is_fallthrough = true;
+        insn_in_progress->addSuccessor(std::move(target), is_call, is_indirect, is_conditional, is_fallthrough);
       }
     }
+
     if(current->op == power_op_bcctr) {
-      insn_in_progress->addSuccessor(makeRegisterExpression(ppc32::ctr), field<31, 31>(insn) == 1,
-                                     true, bcIsConditional, false);
-      if(bcIsConditional) {
-        insn_in_progress->addSuccessor(makeFallThroughExpr(), false, false, false, true);
+      /*********************************************************************
+       *  Power ISA Version 3.1C, Book I. May 2024
+       *  2.4 Branch Instructions
+       *
+       *    bcctr   BO,BI,BH (LK=0)
+       *    bcctrl  BO,BI,BH (LK=1)
+       *
+       *    Extended form examples:
+       *      blctr
+       *      bnectr cr0
+       *
+       *  Programming Note (pg 38)
+       *  ------------------------
+       *
+       *  1. Computed goto’s, case statements, etc.
+       *
+       *     Put the branch address in CTR and set LK=0 (BH=0b11, if appropriate)
+       *
+       *  2. Indirect subroutine linkage
+       *
+       *     Place the final target address in CTR and use bcctr (LK=0).
+       *
+       *     NOTE: This form is not meant to fall through (e.g., it's used in PLT
+       *     thunks), so usually also has BH=0b1z1zz as per Figure 2.5 to indicate
+       *     "branch always".
+       *
+       *  3. Function call
+       *
+       *     For direct calls, put the target address in CTR and use bcctrl (LK=1).
+       ****************************************************************************/
+      bool const LK = field<31, 31>(insn);
+
+      auto add_lr_operand = [this, LK](bool const is_read) {
+        auto link_register = makeRegisterExpression(ppc32::lr);
+        constexpr bool is_implicit = true;
+        bool const is_written = (LK == 1);
+        insn_in_progress->appendOperand(link_register, is_read, is_written, is_implicit);
+      };
+
+      if(LK == 1) {
+        // Case 3 - Function call
+        auto ctr = makeRegisterExpression(ppc32::ctr);
+        constexpr bool is_indirect = true;
+        constexpr bool is_call = true;
+        bool const is_fallthrough = false;
```

Comment:
Any reason why this is not constexpr like the others?
And just a thought, would it be more clear if we define is_indirect/is_call/is_fallthrough/is_conditoinal all as true the start of the function, and then do something like appendOperand(..., is_call, is_indirect, !is_conditiona, !is_fallthrough);

Reply:
> Any reason why this is not constexpr like the others?

Nope, that was just a mistake.

> And just a thought, would it be more clear if we define is_indirect/is_call/is_fallthrough/is_conditoinal all as true the start of the function, and then do something like appendOperand(..., is_call, is_indirect, !is_conditiona, !is_fallthrough);

Good idea! I should do that since I did that in the tests.

Reply:
Updated.